### DISCUSSION TITLE
1주차 사전과제, 김민서

### DISCUSSION THREAD

**Discussion by Sunnystela** [MrMirror21/2024-KHU-CSE-TrackStudy-Web#18] [General]
- 명명 규칙(Naming Convention)이라는 것이 있습니다. 어떤 규칙들이 있는지, 각 규칙들은 어떻게 코드를 작성하는지, 자신이 그동안 사용해왔던 명명 규칙은 무엇인지 적어주세요.
1. Camel case
파일, 변수, 함수 등 대상의 이름을 띄어쓰기 없이 짓는 방법
2. Snake case
단어와 단어 사이를 언더바( _ )를 사용하여 띄워서 이름 짓는 방법
3. Kebab case
단어와 단어 사이를 하이픈( - )을 사용하여 띄워서 이름 짓는 방법

- 함수가 순수하다는 것은 무엇을 의미할까요?
동일한 입력에 대해 항상 동일한 출력을 반환한다. 함수는 입력을 받아 출력을 생성하는 데 그치며, 그 외의 어떤 행동도 수행하지 않는다.

- 개발 시에 순수 함수(pure function)로만 모든 코드를 작성하는 것에 대한 생각을 적어주세요.
순수코드는 출력하는데만 그친다고 했는데 코딩하면 부가적인 작업이 필요한 경우가 많다고 생각한다. 이에 순수코드로만 작성하는 것 보다는 섞어서 사용하는 것이 좋다고 생각한다.

- 자바스크립트에서 함수를 작성하는 문법은 다음과 같습니다. 이 외의 함수를 작성하는 문법 두 가지가 무엇이며, 어떤 때 사용하나요?

```
function myFunc(param) {
	statements
}
```
2. 식으로 작성하기
함수를 변수에 할당하는 방식이다. function키워드를 사용하고 함수의 이름은 변수로 선언된다
```
const Function = myFunc(param){
	statements
}
```
3. 화살표 함수로 작성하기
function 키워드가 필요 없어며 간결한 문법을 제공한다.
```
const myFunc=(param)=>{
	statements
}
```

- 파이썬에서는 문자열을 포맷팅하기 위해 다음과 같은 f-string을 사용합니다. 자바스크립트에서 f-string과 같이 문자열을 다양한 자료형과 변수들을 혼합하여 사용하려면 어떤 방법을 사용하나요?
```
let x = 1;
let y = 2;
console.log(`${x} + ${y}는 ${x + y}입니다.`);
```

-  다음과 같은 객체(object)에서 `p` 값을 출력하려면 `o.p` 와 같은 방식으로 출력해야 하는데, `p` 로 출력할 수 있는 방법은 없을까요?
   ```python
var o = { p: 42, q: true };

console.log(o.p); // 42
console.log(p); // p is undefined
```

```
var o = { p: 42, q: true };

var { p } = o;

console.log(p); // 42
```

**Comment by MrMirror21** [comment]
답변해주신 사전 과제 잘 봤습니다.
전체적으로 잘 답변해주셨고, 아마 답변 입력 하는 과정에서 코드 입력 양식이 좀 틀어진 것 같네요. 다음 번 과제 진행 시에는 이 부분 확인해주시면 좋을 것 같습니다.
